aodndata/acorn/handler.py

```diff
@@ -0,0 +1,133 @@
+                                (?P<nc_time_cov_start>[0-9]{8}T[0-9]{6}Z)_
+                                (?P<platform_code>[A-Z]{3,4})_
+                                (?P<file_version>FV0[0-1])_
+                                (?P<product_type>.*)
+                                \.nc$
+                                """, re.VERBOSE)
+
+
+def get_type(filepath):
+    """return acorn_file_type, the file type of an ACORN file based on its filename"""
+    file_basename = os.path.basename(filepath)
+    unknown_product = False
+    if ACORN_FILE_PATTERN.match(file_basename):
+        fields = get_pattern_subgroups_from_string(file_basename, ACORN_FILE_PATTERN)
+        product_type = fields['product_type']
+        file_version = fields['file_version']
+        platform_code = fields['platform_code']
+
+        if product_type == 'radial' and file_version == 'FV00':
+            acorn_file_type = "radial"
+
+        elif product_type == 'radial' and file_version == 'FV01':
+            acorn_file_type = "radial_quality_controlled"
+
+        elif product_type == 'sea-state' and file_version == 'FV00':
+            acorn_file_type = "vector"
+
+        elif product_type == 'wavespec' and file_version == 'FV01':
+            acorn_file_type = "gridded_1h-avg-wave-spectra_QC"
+
+        elif product_type == 'windp' and file_version == 'FV01':
+            acorn_file_type = "gridded_1h-avg-wind-map_QC"
+
+        elif product_type == 'wavep' and file_version == 'FV01':
+            site_map_station = ['CBG', 'SAG', 'ROT', 'COF']
+
+            if any(s == platform_code for s in site_map_station):
+                acorn_file_type = "gridded_1h-avg-wave-site-map_QC"
+            else:
+                acorn_file_type = "gridded_1h-avg-wave-station-map_QC"
+
+        elif product_type == '1-hour-avg' and file_version == 'FV00':
+            acorn_file_type = "gridded_1h-avg-current-map_non-QC"
+
+        elif product_type == '1-hour-avg' and file_version == 'FV01':
+            acorn_file_type = "gridded_1h-avg-current-map_QC"
+
+        else:
+            unknown_product = True
+    else:
+        unknown_product = True
+
+    if unknown_product:
+        raise InvalidFileNameError("file name: \"{filename}\" Unknown product type from filename".
+                                   format(filename=os.path.basename(filepath)))
+
+    return acorn_file_type
+
+
+class AcornHandler(HandlerBase):
+    def __init__(self, *args, **kwargs):
+        super(AcornHandler, self).__init__(*args, **kwargs)
+        self.allowed_extensions = ['.nc']
+        self.opendap_root = self.config.pipeline_config['global'].get('opendap_root')
+
+    def preprocess(self):
+        nc_collection = self.file_collection[0]
+
+        # enforce a check on the include_regexes and the filename to make sure the handler is processing the correct
+        # matching rexep file written in the config file
+        m = re.search(self.include_regexes[0], nc_collection.name)
```

Comment:
This should be done using core functionality, by using the "allowed_regexes" feature:

https://aodn.github.io/python-aodncore/aodncore.pipeline.html?highlight=handlerbase#aodncore.pipeline.HandlerBase